apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/create-batch/_components/CreateBatchEvaluationModal/useRunBatch.ts

```diff
@@ -0,0 +1,61 @@
+import { useCallback } from 'react'
+
+import { DocumentVersion } from '@latitude-data/core/browser'
+import { runBatchEvaluationAction } from '$/actions/evaluations/runBatch'
+import useLatitudeAction from '$/hooks/useLatitudeAction'
+
+export type RunBatchParameters = Record<string, number | undefined>
+export function useRunBatch({
+  document,
+  projectId,
+  commitUuid,
+  onSuccess,
+}: {
+  projectId: string
+  document: DocumentVersion
+  commitUuid: string
+  onSuccess: () => void
+}) {
+  const {
+    error,
+    execute: run,
+    isPending: isRunning,
+  } = useLatitudeAction(runBatchEvaluationAction, { onSuccess })
+  const errors = error?.fieldErrors
+  const runBatch = useCallback(
+    async ({
+      evaluationIds,
+      wantAllLines,
+      datasetId,
+      parameters,
+      fromLine,
+      toLine,
+    }: {
+      datasetId: number | undefined
+      fromLine: number | undefined
+      toLine: number | undefined
+      wantAllLines: boolean
+      evaluationIds: number[]
+      parameters: RunBatchParameters
+    }) => {
+      await run({
+        projectId: Number(projectId),
+        documentUuid: document.documentUuid,
+        commitUuid,
+        evaluationIds,
+        datasetId: datasetId!,
+        fromLine: wantAllLines ? undefined : fromLine,
+        toLine: wantAllLines ? undefined : toLine,
+        // @ts-ignore
```

Comment:
what happened here

Reply:
Because the form can produce undefined values, I have to lie to server types. Although I could relax server types and do validation as I do

Reply:
If I make the number optional, I can remove this comment. I'll do because is validated with zod anyway.
![image](https://github.com/user-attachments/assets/70ca3bb4-b2df-439b-8863-d2138ffb0362)